public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx

```diff
@@ -196,3 +233,72 @@ export function getPanelLinksBehavior(panel: PanelModel) {
     panelLinksMenu.setState({ links });
   };
 }
+
+function createExtensionContext(panel: VizPanel, dashboard: DashboardScene): PluginExtensionPanelContext {
+  const timeRange = sceneGraph.getTimeRange(panel);
+  let queryRunner = panel.state.$data;
+  let targets: DataQuery[] = [];
+  const id = getPanelIdForVizPanel(panel);
+
+  if (queryRunner instanceof SceneDataTransformer) {
+    queryRunner = queryRunner.state.$data;
+  }
+
+  if (queryRunner instanceof SceneQueryRunner) {
+    targets = queryRunner.state.queries;
+  }
+
+  if (queryRunner instanceof ShareQueryDataProvider) {
+    targets = [queryRunner.state.query];
+  }
+
+  let scopedVars = {};
+
+  // Handle panel repeats scenario
+  if (panel.state.$variables && panel.state.$variables instanceof SceneVariableSet) {
```

Comment:
you don't need `panel.state.$variables instanceof SceneVariableSet`, the `$variables` SceneVariables interface has the state type you need here to just get the variables

Reply:
yup, that's tru